pkg/monitortests/node/kubeletlogcollector/monitortest.go

```diff
@@ -60,3 +64,39 @@ func (*kubeletLogCollector) Cleanup(ctx context.Context) error {
 	// TODO wire up the start to a context we can kill here
 	return nil
 }
+
+func nodeFailedLeaseErrors(finalIntervals monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "[sig-node] kubelet-log-collector detects node failed to lease events"
+	var failures []string
+	var firstEventForE2ETest time.Time
+	// Let's only detect FailedLeaseError once the e2e tests are starting.
+	for _, event := range finalIntervals {
+		if event.Message.Reason == monitorapi.E2ETestStarted {
+			firstEventForE2ETest = event.From
+		}
+	}
+	for _, event := range finalIntervals {
+		if event.Message.Reason == monitorapi.NodeFailedLease {
+			if event.From.After(firstEventForE2ETest) {
+				failures = append(failures, fmt.Sprintf("%s %v - %v", event.From.Format(time.RFC3339), event.Locator.OldLocator(), event.Message.OldMessage()))
+			}
+		}
+	}
+
+	// failures during a run always fail the test suite
+	var tests []*junitapi.JUnitTestCase
+	if len(failures) > 0 {
+		tests = append(tests, &junitapi.JUnitTestCase{
+			Name:      testName,
+			SystemOut: strings.Join(failures, "\n"),
+			FailureOutput: &junitapi.FailureOutput{
+				Output: fmt.Sprintf("kubelet-log-collector reports %d node failed to lease events.\n\n%v", len(failures), strings.Join(failures, "\n")),
+			},
+		})
+	}
+
+	if len(tests) == 0 {
+		tests = append(tests, &junitapi.JUnitTestCase{Name: testName})
```

Comment:
let's only flake if we don't hav emore than two lease failures in 33 seconds on a single node.

Reply:
updated.